qa/os/src/test/java/org/elasticsearch/packaging/test/KeystoreManagementTests.java

```diff
@@ -0,0 +1,416 @@
+    }
+
+    public void test50KeystorePasswordFromFile() throws Exception {
+        assumeTrue("only for systemd", Platforms.isSystemd() && distribution().isPackage());
+        String password = "!@#$%^&*()|\\<>/?";
+        Path esKeystorePassphraseFile = installation.config.resolve("eks");
+
+        rmKeystoreIfExists();
+        createKeystore();
+        setKeystorePassword(password);
+
+        assertPasswordProtectedKeystore();
+
+        try {
+            sh.run("sudo systemctl set-environment ES_KEYSTORE_PASSPHRASE_FILE=" + esKeystorePassphraseFile);
+
+            Files.createFile(esKeystorePassphraseFile);
+            Files.write(esKeystorePassphraseFile,
+                (password + System.lineSeparator()).getBytes(StandardCharsets.UTF_8),
+                StandardOpenOption.WRITE);
+
+            startElasticsearch();
+            ServerUtils.runElasticsearchTests();
+            stopElasticsearch();
+        } finally {
+            sh.run("sudo systemctl unset-environment ES_KEYSTORE_PASSPHRASE_FILE");
+        }
+    }
+
+    public void test51WrongKeystorePasswordFromFile() throws Exception {
+        assumeTrue("only for systemd", Platforms.isSystemd() && distribution().isPackage());
+        Path esKeystorePassphraseFile = installation.config.resolve("eks");
+
+        assertPasswordProtectedKeystore();
+
+        try {
+            sh.run("sudo systemctl set-environment ES_KEYSTORE_PASSPHRASE_FILE=" + esKeystorePassphraseFile);
+
+            if (Files.exists(esKeystorePassphraseFile)) {
+                rm(esKeystorePassphraseFile);
+            }
+
+            Files.createFile(esKeystorePassphraseFile);
+            Files.write(esKeystorePassphraseFile,
+                ("wrongpassword" + System.lineSeparator()).getBytes(StandardCharsets.UTF_8),
+                StandardOpenOption.WRITE);
+
+            Shell.Result result = runElasticsearchStartCommand();
+            assertElasticsearchFailure(result, PASSWORD_ERROR_MESSAGE);
+        } finally {
+            sh.run("sudo systemctl unset-environment ES_KEYSTORE_PASSPHRASE_FILE");
+        }
+    }
+
+    /**
+     * Check that we can mount a password-protected keystore to a docker image
+     * and provide a password via an environment variable.
+     */
+    public void test60DockerEnvironmentVariablePassword() throws Exception {
+        assumeTrue(distribution().isDocker());
+        String password = "password";
+        Path dockerKeystore = installation.config("elasticsearch.keystore");
+
+        Path localKeystoreFile = getKeystoreFileFromDockerContainer(password, dockerKeystore);
+
+        // restart ES with password and mounted keystore
+        Map<Path, Path> volumes = Map.of(localKeystoreFile, dockerKeystore);
+        Map<String, String> envVars = Map.of("KEYSTORE_PASSWORD", password);
+        runContainer(distribution(), volumes, envVars);
+        waitForElasticsearch(installation);
+        ServerUtils.runElasticsearchTests();
```

Comment:
Is it not sufficient to just ensure that Elasticsearch has started? Do we need to also run the tests?

Reply:
I don't see the downside in checking that Elasticsearch is not just running but minimally functional as well.